frontend/src/components/shortener-input/handleShortener.ts

```diff
@@ -82,6 +83,33 @@ export async function handleShortener({ state }: any) {
   confettiAnimate();
 }
 
+/**
+ * Share on social media
+ * @param {String} socialMedia - Social network name
+ */
+export function shareOnSocialMedia(socialMedia:string) {
+  const shorten_url = document.querySelector('#result #text')!.textContent;
+  if(!shorten_url) return
+  switch (socialMedia) {
+    case "twitter":
+      let base_url = "https://twitter.com/share?";
+      const params:ISocialMedia = {
+        text: "My url has just reduced!\nCheck out:",
+        url: shorten_url ,
+        hashtags: "reducedto,shortenurl",
+      }
+      for(const prop in params) {
+        const index_value = params[`${prop}`]
+        if(!index_value) continue
+        base_url += `&${prop}=${encodeURIComponent(index_value)}`
+      }
```

Comment:
Why do we need this for loop?

Reply:
because the values ​​are defined in the 'params' object and it is required to concatenate each one to the base url string.
The object is typed by the possible values ​​that the url receives, for example currently the type 'via' is not being used, but when required it is only added to the object or a new object is passed as a parameter, the for loop finds it and concatenates it.
do you agree? or just concatenate the possible values ​​as string
`let base_url = "https://twitter.com/share?url=shorten_url.com&text=My%20url%20has%20just%20reduced!%5CnCheck%20out%3A&hashtags=reducedto%2Cshortenurl";`

Reply:
I am pretty sure you can build the url like this:
```suggestions
http://twitter.com/share?text=text goes here&url=https://reduced.to goes here&hashtags=hashtag1,hashtag2,hashtag3
```

Good  luck 🚀